types/float64.go

```diff
@@ -0,0 +1,86 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func float64ValueFromTerraform(ctx context.Context, in tftypes.Value) (attr.Value, error) {
+	if !in.IsKnown() {
+		return Float64{Unknown: true}, nil
+	}
+
+	if in.IsNull() {
+		return Float64{Null: true}, nil
+	}
+
+	var bigF *big.Float
+	err := in.As(&bigF)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Validation is handled by Validate method.
+	f, _ := bigF.Float64()
```

Comment:
I'd say let's still bubble up the error anyways, for when things don't work the way they're supposed to. Error checking in depth. :)

Reply:
You got it!

Reply:
Ah, I'm not positive of this, but an interesting caveat to this is that this error I think will always be returned first instead of the `Validate()` diagnostics. I guess the only real difference is the error messaging/formatting, but could be something to consider.

Reply:
Will it? I thought we designed `Validate` diagnostics to run _before_ trying to create any `attr.Value`s, which is (as far as I can tell) where this gets used? Otherwise, the `Validate` would take an `attr.Value` argument, I think?

Reply:
https://github.com/hashicorp/terraform-plugin-framework/blob/84afb9afc9ce3e8e4777ca9024bd641ff646c363/tfsdk/config.go#L46-L73 This seems to suggest to me, for GetAttribute on Config specifically at least, that the `Validate` diags will be returned instead of the `ValueFromTerraform` error when both are returned. I think the reason I'm arguing we should handle it here anyways is as a safeguard in case we ever forget to call `Validate` or call them in the wrong order or something.